gateway/api/access_policies/jobs.py

```diff
@@ -0,0 +1,67 @@
+"""
+Access policies implementation for Job access
+"""
+import logging
+from django.contrib.auth import get_user_model
+from api.models import Job
+
+User = get_user_model()
+
+
+logger = logging.getLogger("gateway")
+
+
+class JobAccessPolocies:  # pylint: disable=too-few-public-methods
+    """
+    The main objective of this class is to manage the access for the user
+    to the Job entities.
+    """
+
+    @staticmethod
+    def can_access(user: User, job: Job) -> bool:
+        """
+        Checks if the user has access to save the result of a Job:
+
+        Args:
+            user: Django user from the request
+            job: Job instance against to check the access
+
+        Returns:
+            bool: True or False in case the user has access
+        """
+
+        is_provider_job = job.program and job.program.provider
+        if is_provider_job:
+            provider_groups = job.program.provider.admin_groups.all()
+            author_groups = user.groups.all()
+            has_access = any(group in provider_groups for group in author_groups)
```

Comment:
Here @paaragon you can use the `Provider.can_access`: https://github.com/Qiskit/qiskit-serverless/blob/main/gateway/api/access_policies/providers.py#L19

I think it does exactly the same thing that you have in those 3 lines of code. It checks if the user has access to that provider.